Add sanity checks to CI for clusterfuzz release

everestmz opened the pull request:
cc: @google/clusterfuzz 

This CL adds the following functionality to the CI module:
 - Clones chromium & installs depot tools
 - Installs the latest release candidate (hardcoded right now)
 - Runs all testcases inside sanity_checks.yml
 - Logs the failure or success to Stackdriver
 - Adds a new service, `auth-update`, that keeps the auth token updated so it doesn't expire in the middle of a testcase run
 - Various updates to `playbook.yml` to make everything work, like using a 100GB disk size, moving to ubuntu, ensuring that clusterfuzz user has password-less sudo, etc

tanin47 commented on .gitignore:
```diff
@@ -1,5 +1,6 @@
 *.egg-info
 *.pyc
+*.py~
```
What is this file?

everestmz replied:
Sometimes vim doesn't clean up it's swap files properly, this just ignores them.

tanin47 replied:
Let's remove it; it's oddly specific to support only vim. (I point my swap file to `tmp`).

tanin47 commented on ci/auth_update/main.py:
```diff
@@ -0,0 +1,31 @@
+"""Keeps the authorization token in ~/.clusterfuzz/auth_header updated."""
```
Why do we need to have a separate process for this? Specifically, why doesn't a method in main.py work?

tanin47 commented on ci/continuous_integration/ci/main.py:
```diff
@@ -0,0 +1,80 @@
+    if os.path.exists(CLUSTERFUZZ_BUILD):
+      shutil.rmtree(CLUSTERFUZZ_BUILD)
+    stackdriver_logging.send_run(
+        testcase, 'sanity', '0.2.2rc3', run_testcase(testcase))
```
I think we should use a decoration to capture any exception from it.

everestmz replied:
What would the decorator do? The exception would be recorded by svlogd and runit would restart the process so we don't have to worry about that.

tanin47 replied:
Ok, fair enough. This is good then.

tanin47 commented on circle.yml:
```diff
@@ -6,6 +6,6 @@ test:
     - cd tool && coverage run test.py
+    - cd ci/continuous_integration && coverage run test.py
   post:
     - cd tool && coveralls
```
This doesn't send coverage of ci/continuous_integration to coveralls. Need more research on this part.

tanin47 commented on ci/continuous_integration/test/helpers.py:
```diff
@@ -0,0 +1,85 @@
+"""Helper methods and classes to be used by all tests."""
```
I think we should refactor this and share the code between `tool` and `ci`. For tests, we can manually modify `sys.path` in the code.

tanin47 commented on ci/continuous_integration/ci/clone_chromium.py:
```diff
@@ -0,0 +1,35 @@
+  """Runs the correct commands to clone chromium & depot tools."""
+
+  if not os.path.exists(DEPOT_TOOLS):
+    subprocess.call([
```
Do `shell=True` on all of them, so that they are in the same format (nicer) and is closer to what humans do.

tanin47 commented on ci/continuous_integration/ci/clone_chromium.py:
```diff
@@ -0,0 +1,35 @@
+  subprocess.call('export PATH=$PATH:%s && gclient runhooks' % DEPOT_TOOLS,
+                  shell=True, cwd=CHROMIUM_SRC)
+
+def main():
```
Remove this `main()`. We don't seem to need it. Move `if` into `clone_chromium`.

tanin47 commented on ci/continuous_integration/ci/clone_chromium.py:
```diff
@@ -0,0 +1,35 @@
+    clone_chromium()
+
+if __name__ == '__main__':
+  main()
```
Just call `clone_chromium` directly. Normally, we have `main` because its responsibility is to parse flag.

everestmz commented:
Updated:
 - removed __main__ functionality from clone_chromium
 - moved helpers.py into shared folder and modified sys.path in both test.py
 - moved updating auth into the ci module
 - small changes

Still need to work out sending coverage for both test results

tanin47 commented on ci/playbook.yml:
```diff
@@ -3,13 +3,26 @@
+        image: ubuntu-1404
+        name: "ci-test-boot"
+        state: present
+        service_account_email: "ci-test-tanin@clusterfuzz-tools.iam.gserviceaccount.com"
```
After this CL, please switch to `ci-test@` and move the credentials file to `/auto/...`

tanin47 commented on ci/playbook.yml:
```diff
@@ -3,13 +3,26 @@
         instance_names: continuous-integration-test
-        machine_type: n1-standard-8
-        image: debian-7
+        machine_type: n1-standard-32
```
64 is not possible?

everestmz replied:
Not in the current region. I didn't want to switch regions along with all of this - I'll do it in another CL

tanin47 replied:
sounds good.

tanin47 commented on ci/playbook.yml:
```diff
@@ -65,22 +94,23 @@
-        - {'src': 'python_daemon_run', 'dest': '/etc/sv/python-daemon/run'}
-        - {'src': 'python_daemon_log_run', 'dest': '/etc/sv/python-daemon/log/run'}
-        - {'src': 'test.py', 'dest': '/python-daemon/test.py'}
+        - {'src': 'python_daemon/run', 'dest': '/etc/sv/python-daemon/run'}
```
Do we still need this python_daemon?
